### DISCUSSION TITLE
Recursive parse with right array-style indexing

### DISCUSSION THREAD

**Discussion by 0x0015** [foonathan/lexy#201] [Q&A] [answered]
Hello,

I'm trying to parse a type (somewhat similar to C) where there are builtin types, tuple types, or array types.

The complication is for the array type to be parsed `/*type*/ [*/number*/]`, but allow recursion (eg. `int[5][7]` is parsed as `(int[5])[7]`)

Is there any way to do this?
I have a simplified example that is very clear of what I'm trying to do here: https://lexy.foonathan.net/playground/?id=qExhjxKPY&mode=trace
and my best attempt here: https://lexy.foonathan.net/playground/?id=K3b3hahGe&mode=trace (though it's a basically a hack to get it to parse, but much is lost)

The best way I could think to do it would be to somehow consume the square bracketed part from the right, and parse the rest on the left (not including that) recursively, but I don't know how that could be done in lexy.

**Comment by foonathan** [comment, marked as answer]
You want to treat an array as a postfix operator: https://lexy.foonathan.net/playground/?id=TdWqK7adK&mode=trace
